### ISSUE TITLE
Feature: Have the possibility to create Sensors and Conditions with parameters

### ISSUE THREAD

**Issue by Racines** [crashkonijn/GOAP#250]
When we link a sensor or a condition in the CapabilityBuilder, it would be nice to have an equivalent of IActionProperties or the possibility to link instance instead of generic type.

**Comment by keiranlovett** [comment]
+1

This would greatly increase the adaptability.

**Comment by crashkonijn** [comment]
@Racines I can understand the sensors, but can you give me an example of what you'd like to do with the conditions?

**Comment by Racines** [comment]
maybe what I mean is for Local/World Keys, effects, actions, and sensors instead of condition.
The exemple would be something like this:

```csharp
public enum EEatableType
{
	Apple,
	Orange,
	Banana,
	
	Count,
}

public record Eatable(EEatableType Type, float Cost);


public void AddActionForEatable(Builder builder, Eatable eatable)
{	
	var eatableType = eatable.Type;

	var pickupAction = new PickupItemAction<IEatable>(eatableType)
	var sensorKey = new IsInWorld<IEatable>(eatableType);
	var effectKey = new IsHolding<IEatable>(eatableType);
	var targetKey = new ClosestTarget<IEatable>(eatableType);
	var targetSensor = new ClosestItemSensor<IEatable>(eatableType);
	var worldSensor = new IsInWorldSensor<IEatable>(eatableType);
	
	builder.AddAction(pickupAppleAction)
		.SetTarget(targetKey)
		.AddEffect(effectKey, true)
		.AddCondition(sensorKey , Comparison.GreaterThanOrEqual, 1)
		.SetBaseCost(eatable.Cost)
		.SetInRange(0.3f);


	builder.AddTargetSensor(targetSensor)
		.SetTarget(targetKey);

	 builder.AddWorldSensor(worldSensor)
		.SetKey(sensorKey);
}

public void AddAllEatableActions(Builder builder)
{
	var eatables = new List<Eatable>()
	{
		new (EEatableType.Apple, 1),
		new (EEatableType.Orange, 2),
		new (EEatableType.Banana, 4),
	};
	
	foreach (var eatable in eatables)
		AddActionForEatable(builder, eatable);
}
```

**Comment by crashkonijn** [comment]
Fixed with 3.0.20-beta!
